jobbergate-api/jobbergate_api/apps/job_scripts/job_script_files.py

```diff
@@ -0,0 +1,269 @@
+            check(
+                default_template_name in application_files.templates,
+                f"{default_template_name=} was not found",
+            )
+            if app_config.jobbergate_config.supporting_files_output_name:
+                for supporting_file in app_config.jobbergate_config.supporting_files_output_name.keys():
+                    check(supporting_file in application_files.templates, f"{supporting_file=} was not found")
+
+        with JobScriptCreationError.handle_errors("Error rendering the main file"):
+
+            param_dict_flat = flatten_param_dict(app_config.dict())
+
+            main_file_path = Path(
+                app_config.jobbergate_config.output_directory,
+                default_template_name.rstrip(".j2").rstrip(".jinja2"),
+            )
+
+            main_file_content = render_template(
+                application_files.templates[default_template_name],
+                param_dict_flat,
+            )
+
+        with JobScriptCreationError.handle_errors("Error while injecting the sbatch params"):
+            if sbatch_params:
+                main_file_content = inject_sbatch_params(main_file_content, sbatch_params)
+
+        with JobScriptCreationError.handle_errors("Error while creating JobScriptFiles object"):
+            jobscript_files = cls(
+                main_file_path=main_file_path,
+                files={main_file_path: main_file_content},
+            )
+
+        with JobScriptCreationError.handle_errors("Error while rendering the supporting files"):
+            output_name_mapping = app_config.jobbergate_config.supporting_files_output_name
+            if output_name_mapping:
+                for template_name, supporting_filename_list in output_name_mapping.items():
+
+                    for supporting_filename in supporting_filename_list:
+                        path = Path(
+                            app_config.jobbergate_config.output_directory,
+                            supporting_filename,
+                        )
+
+                        jobscript_files.files[path] = render_template(
+                            application_files.templates[template_name],
+                            param_dict_flat,
+                        )
+
+        logger.debug("Done rendering job-script files from an application")
+
+        return jobscript_files
+
+    @classmethod
+    def file_manager_factory(self, job_script_id: int) -> FileManager:
+        """
+        Build an application file manager.
+        """
+        return cast(
+            FileManager,
+            file_manager_factory(
+                id=job_script_id,
+                s3_client=s3_client,
+                bucket_name=settings.S3_BUCKET_NAME,
+                work_directory=Path(JOBSCRIPTS_WORK_DIR),
+                manager_cls=FileManager,
+                transformations=IO_TRANSFORMATIONS,
+            ),
+        )
+
+
+def render_template(template_file, params):
```

Comment:
This seems like a convenience method that could just be replaced with `Template(template_file).render(data=params)`. Since you aren't doing any error checks and it's just two consecutive function calls that can be inlined, I would recommend dropping this one.

Reply:
Solved in 99cb81ef9a1a7ff8af7d637953daead66dcf65ea.